horovod/mxnet/handle_manager.cc

```diff
@@ -43,8 +43,16 @@ bool HandleManager::PollHandle(int handle) {
 }
 
 void HandleManager::ExecuteCallback(int handle) {
-  Callback cb = *callbacks_[handle];
-  cb();
+  std::unique_lock<std::mutex> lock(mutex_);
+  if (callbacks_.find(handle) == callbacks_.end()) {
+    return;
+  }
+  auto cb_ptr = callbacks_[handle];
```

Comment:
I would actually prefer the Callback data type here to increase readability

Reply:
It is easier to do nullptr check in this way.